Neos.RedirectHandler/Classes/Command/RedirectionCommandController.php

```diff
@@ -0,0 +1,336 @@
+use TYPO3\Flow\Cli\CommandController;
+use TYPO3\Flow\Persistence\PersistenceManagerInterface;
+use TYPO3\Flow\Utility\Arrays;
+
+/**
+ * Command controller for tasks related to redirects
+ *
+ * @Flow\Scope("singleton")
+ */
+class RedirectionCommandController extends CommandController
+{
+    /**
+     * @Flow\Inject
+     * @var RedirectStorageInterface
+     */
+    protected $redirectStorage;
+
+    /**
+     * @Flow\Inject
+     * @var PersistenceManagerInterface
+     */
+    protected $persistenceManager;
+
+    /**
+     * @param string $host Filter redirection by the given hostname
+     * @param string $match Match source or target URI by a regular expression
+     * @return void
+     */
+    public function listCommand($host = null, $match = null)
+    {
+        $outputByHost = function ($host) use ($match) {
+            $redirects = $this->redirectStorage->getAll($host);
+            $this->outputLine();
+            if ($host !== null) {
+                $this->outputLine('<info>==</info> <b>Redirection for %s</b>', [$host]);
+            } else {
+                $this->outputLine('<info>==</info> <b>Redirection without host attached</b>', [$host]);
+            }
+            if ($match !== null) {
+                $this->outputLine('   <info>++</info> <b>Show only if source or target URI match <u>%s</u></b>', [$match]);
+                sleep(1);
+            }
+            $this->outputLine();
+            /** @var $redirect RedirectInterface */
+            foreach ($redirects as $redirect) {
+                $outputLine = function ($source, $target, $statusCode) {
+                    $this->outputLine('   <comment>></comment> %s <info>=></info> %s <comment>(%d)</comment>', [
+                        $source,
+                        $target,
+                        $statusCode
+                    ]);
+                };
+                $source = $redirect->getSourceUriPath();
+                $target = $redirect->getTargetUriPath();
+                $statusCode = $redirect->getStatusCode();
+                if ($match === null) {
+                    $outputLine($source, $target, $statusCode);
+                } else {
+                    $regexp = sprintf('#%s#', $match);
+                    $matches = preg_grep($regexp, [$target, $source]);
+                    if (count($matches) > 0) {
+                        $replace = "<error>$0</error>";
+                        $source = preg_replace($regexp, $replace, $source);
+                        $target = preg_replace($regexp, $replace, $target);
+                        $outputLine($source, $target, $statusCode);
+                    }
+                }
+            }
+        };
+        if ($host !== null) {
+            $outputByHost($host);
+        } else {
+            $hosts = $this->redirectStorage->getDistinctHosts();
+            if ($hosts !== []) {
+                array_map($outputByHost, $hosts);
+            }
+            $outputByHost($host);
```

Comment:
Should be replaced by ```$outputByHost();```, will push this later today

But a redirect can be attached to a host or not and outputByHost handle correctly null value.